client/src/client_proxy.rs

```diff
@@ -422,6 +454,59 @@ impl ClientProxy {
         )
     }
 
+    /// Transfers coins from sender to receiver.
+    pub fn prepare_transfer_coins(
```

Comment:
Perhaps I'm missing it somewhere, but it seems that this isn't actually being used anywhere?

Reply:
It is exposed as an API function on the `ClientProxy` struct.
This is useful when using `client` module as a library.
You can see a test case in https://github.com/libra/libra/pull/438/files#diff-6d190f5f4aeb4c850e66a4eb2267e5d3R285.

Reply:
Can you provide a bit more context about when you expect to use this?  And why we pass in space_delim_strings when this isn't being called through the CLI?  the space delim strings come from the parameters passed to the CLI - if we aren't passing through CLI, then we should just pass normal args

Reply:
Sure: in my specific application we use a threshold signature scheme (TSS) to sign a given unsigned raw transaction, and then attach the result signature to the raw transaction and broadcast it. 
Meaning, instead of using the private key generated by default with the Libra wallet, we use an _external_ scheme where different parties collaboratively compute a joint signature.
This is useful in general for any external source of signing, e.g., a hardware wallet. 
Such external signers are usually agnostic by design to the underlying transaction semantics, and instead just sign a given hash for a desired signing algorithm (Ed25519 in our case).
---
I pushed modifications to my code so the methods would accept the normal args instead of space delimited strings (originally I used them inside the CLI app and then decided it would be cleaner to use as API methods alone).